x-pack/auditbeat/module/system/host/host.go

```diff
@@ -26,25 +30,64 @@ func init() {
 // MetricSet collects data about the host.
 type MetricSet struct {
 	mb.BaseMetricSet
+	log *logp.Logger
 }
 
 // New constructs a new MetricSet.
 func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	cfgwarn.Experimental("The %v/%v dataset is experimental", moduleName, metricsetName)
 
-	config := defaultConfig
+	config := config.NewDefaultConfig()
 	if err := base.Module().UnpackConfig(&config); err != nil {
 		return nil, errors.Wrapf(err, "failed to unpack the %v/%v config", moduleName, metricsetName)
 	}
 
-	return &MetricSet{base}, nil
+	ms := &MetricSet{
+		BaseMetricSet: base,
+		log:           logp.NewLogger(moduleName),
+	}
+
+	if config.ReportChanges {
+		// TODO: Implement reporting changes?
+		ms.log.Warnw("Metricset %v/%v does not support report_changes", moduleName, metricsetName)
```

Comment:
Makes sense, and I see the `cpu` and `core` metricsets of the `system` module in Metricbeat do that as well. I wonder if the configuration is counterintuitive - I myself tried at first to nest it under the metricset i.e.:

```
- module: system
  metricsets:
    - processes
      report_changes: true
```

Should we consider allowing something like this in the future, rather than have it apart?

For now, I'll change it to be `processes.report_changes` and `packages.report_changes`.